backend/cache/redis.go

```
@@ -36,46 +37,77 @@ func NewRedisCache(port string, config config.CacheConfigProvider) CacheProvider
 }
 
 // Set `value` for specified `key`
-func (cache *RedisCache) Set(key string, value interface{}) {
-	err := cache.client.Set(key, value, cache.config.GetCacheExpirationTime()).Err()
+func (cache *RedisCache) Set(key string, value interface{}) error {
+	bytes, err := msgpack.Marshal(value)
+	if err != nil {
+		log.WithFields(log.Fields{
+			"key":   key,
+			"value": value,
+		}).WithError(err).Error("set: failed to marshal value")
+		return err
+	}
+
+	err = cache.client.Set(key, bytes, cache.config.GetCacheExpirationTime()).Err()
 	if err != nil {
 		log.WithFields(log.Fields{
 			"key":   key,
 			"value": value,
 		}).WithError(err).Error("set: failed to set key and value in cache")
+		return err
 	}
+
+	return nil
 }
 
 // Set `value` for specified key where key is created by hashing `fields`
-func (cache *RedisCache) SetWithFields(fields Fields, value interface{}) {
-	cache.Set(convertFieldsToKey(fields), value)
+func (cache *RedisCache) SetWithFields(fields Fields, value interface{}) error {
+	return cache.Set(convertFieldsToKey(fields), value)
 }
 
 // Get value for specified `key`
-func (cache *RedisCache) Get(key string) (interface{}, bool) {
-	result, err := cache.client.Get(key).Result()
-	if err != nil {
+func (cache *RedisCache) Get(key string, value interface{}) (bool, error) {
```

Side effects are strong with this one!
Maybe (we ignore the error of this function anyway) change the interface to standard library map-like? That would look something like: Get(key string) (interface{}, bool) and then you call it with value, ok := cache.Get(key)?